### PR TITLE
[Payment] Clean up payment state machine

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

**Comment by GSadee** [comment]
Done 👍

File: UPGRADE.md [hunk 1]
```diff
@@ -200,9 +200,14 @@ After:
 * Credit Card model and all related code have been removed.
 * `->getSource()` and `->setSource(PaymentSourceInterface $source)` have been removed from `PaymentInterface`.
 * `PaymentSourceInterface` has been removed.
+* *void* transition and state has been removed due to being no used.
 
 ### PayumBundle
 
+* There was changes with handling payment states:
```
Comment (pamil) [inline UPGRADE.md]: "There was changes" -> "There **were** changes **made**"?

File: UPGRADE.md [hunk 2]
```diff
@@ -200,9 +200,14 @@ After:
 * Credit Card model and all related code have been removed.
 * `->getSource()` and `->setSource(PaymentSourceInterface $source)` have been removed from `PaymentInterface`.
 * `PaymentSourceInterface` has been removed.
+* *void* transition and state has been removed due to being no used.
```
Comment (pamil) [inline UPGRADE.md]: `*void*` -> `` `void` ``?
"no" -> "not"

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

**Comment by TheMadeleine** [comment]
Needs an update in the state machine diagram in the docs! :P I'm on it.

**Pull request merged**